x-pack/plugins/cases/public/components/user_actions/helpers.ts

```diff
@@ -23,5 +25,41 @@ export const getManualAlertIdsWithNoRuleId = (comments: Comment[]): string[] =>
     }
     return alertIds;
   }, new Set<string>());
-  return [...dedupeAlerts];
+  return Array.from(dedupeAlerts);
+};
+
+export const getManualAlertIds = (comments: Comment[]): string[] => {
+  const dedupeAlerts = comments.reduce((alertIds, comment: Comment) => {
+    if (comment.type === CommentType.alert) {
+      const ids = Array.isArray(comment.alertId) ? comment.alertId : [comment.alertId];
+      ids.forEach((id) => alertIds.add(id));
+      return alertIds;
+    }
+    return alertIds;
+  }, new Set<string>());
+  return Array.from(dedupeAlerts);
+};
+
+export const getFeatureIdsFromAlertIndices = (comments: Comment[]): ValidFeatureId[] => {
+  const dedupeFeatureId = comments.reduce((featureIds, comment: Comment) => {
+    if (comment.type === CommentType.alert) {
+      const indices = Array.isArray(comment.index) ? comment.index : [comment.index];
+      indices.forEach((index) => {
```

Comment:
Is the name of an index fixed and permanent or it is user/plugin-configurable? If it can change by the plugin is there any constant exported from the plugin we can use?

Reply:
We fixed it by creating a new API to get FeatureIds by registrationContext of the index name